sourcegraph/sourcegraph
# search: Refactor search stack UI widget

This makes some updates to the look and feel of the widget. In
particular it changes the closed state of the widget from a small button
to showing the top item. Clicking on the item will open the widget.

The buttons for adding a new entry will now always be shown, even if the
entry was just added.

Annotations can now be added. Those are rendered as markdown blocks
*before* the references block.

More thought has to be put into showing buttons for adding new entries in the closed state.


https://user-images.githubusercontent.com/179026/154087643-7c7337d9-4450-41be-87c3-205f2408f5fc.mp4




## Test plan

- Go to /search or any a file view. The widget should show up, showing buttons to add new entries.
- Click on the button to add an entry.
- Click on the entry to open the widget.
- Click on the "annotations button" to toggle the annotations input.
- Click on the notebook button. The annotations should be converted to markdown blocks before the search/file block.
